test/e2e/performance_test.go

```diff
@@ -0,0 +1,371 @@
+}
+
+func BenchmarkWorkload10000HTTPRequest100000(b *testing.B) {
+	withPerformanceTestSetup(func(data *TestData) { httpRequest(100000, 10000, data, b) }, b)
+}
+
+func BenchmarkWorkload15000HTTPRequest100000(b *testing.B) {
+	withPerformanceTestSetup(func(data *TestData) { httpRequest(100000, 15000, data, b) }, b)
+}
+
+func BenchmarkRealizeNetworkPolicy5000(b *testing.B) {
+	withPerformanceTestSetup(func(data *TestData) { networkPolicyRealize(5000, data, b) }, b)
+}
+
+func BenchmarkRealizeNetworkPolicy10000(b *testing.B) {
+	withPerformanceTestSetup(func(data *TestData) { networkPolicyRealize(10000, data, b) }, b)
+}
+
+func BenchmarkRealizeNetworkPolicy150000(b *testing.B) {
+	withPerformanceTestSetup(func(data *TestData) { networkPolicyRealize(15000, data, b) }, b)
+}
+
+func randCidr(rndSrc rand.Source) string {
+	return fmt.Sprintf("%d.%d.%d.%d/32", rndSrc.Uint64()%255+1, rndSrc.Uint64()%255+1, rndSrc.Uint64()%255+1, rndSrc.Uint64()%255+1)
+}
+
+// createPerformanceTestPodSpec creates Pod description for the performance test.
+// The pod would be scheduled to the master node.
+func createPerformanceTestPodSpec(name, containerName, image string) *v1.Pod {
+	podSpec := v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:            containerName,
+				Image:           image,
+				ImagePullPolicy: v1.PullIfNotPresent,
+			},
+		},
+		RestartPolicy: v1.RestartPolicyNever,
+	}
+	podSpec.NodeSelector = map[string]string{
+		"kubernetes.io/hostname": masterNodeName(),
+	}
+	// tolerate NoSchedule taint to let the pod run on master node
+	noScheduleToleration := v1.Toleration{
+		Key:      "node-role.kubernetes.io/master",
+		Operator: v1.TolerationOpExists,
+		Effect:   v1.TaintEffectNoSchedule,
+	}
+	podSpec.Tolerations = []v1.Toleration{noScheduleToleration}
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"app": performanceAppLabel},
+		},
+		Spec: podSpec,
+	}
+	return pod
+}
+
+// createNginxPodOnMaster creates the nginx pod which is used for the http test server.
+func createNginxPodOnMaster(data *TestData) error {
+	nginxPod := createPerformanceTestPodSpec(performanceNginxPodName, nginxContainerName, "nginx")
+	if _, err := data.clientset.CoreV1().Pods(testNamespace).Create(nginxPod); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createNginxPodOnMaster creates the ab pod which is used for the http test client.
+func createABPodOnMaster(data *TestData) error {
+	sleepDuration := "3600" // seconds
+	abPod := createPerformanceTestPodSpec(performanceABPodName, abName, abImage)
+	abPod.Spec.Containers[0].Command = []string{"sleep", sleepDuration}
+	_, err := data.clientset.CoreV1().Pods(testNamespace).Create(abPod)
+	return err
+}
+
+// createPerformanceNginx creates the nginx pod and wait it to be ready.
+func createPerformanceNginx(data *TestData, b *testing.B) (string, error) {
+	b.Logf("Creating a nginx test Pod")
+	if err := createNginxPodOnMaster(data); err != nil {
```

Comment:
not sure what's the benefit of splitting this into 2 functions (`createNginxPodOnMaster` and `createPerformanceNginx`)?